internal/pkg/service/buffer/usererror/usererror.go

```diff
@@ -1,4 +1,5 @@
-package service
+// Package usererror provide for distinguishing between user error (expected error) and application error (unexpected error).
+package usererror
```

Comment:
Funkcie nizsie potrebujem na viacerych miestach v buffer kode, tak som na to spravil novy pkg.